server/server.js

```diff
@@ -85,27 +85,43 @@ app.post(`${apiBaseUrl}/form`,
     return res.json({ id, isolationPlanStatus });
   }));
 
-// Create new form, not secured
+// Submission from PHAC, secured by JWT for PHAC users only
 app.post(`${apiBaseUrl}/phac/submission`,
   passport.authenticate('jwt-phac', { session: false }),
   asyncMiddleware(async (req, res) => {
     await validate(PhacSchema, req.body); // Validate submitted submissions against schema
     const phacCollection = dbClient.db.collection(collections.PHAC);
-    const id = await generateUniqueHexId(phacCollection);
 
     const currentIsoDate = new Date().toISOString();
-    const phacItems = req.body.map((item) => ({
+    const phacItems = await Promise.all(req.body.map(async (item) => ({
       ...item,
-      id,
+      id: await generateUniqueHexId(phacCollection),
       createdAt: currentIsoDate,
       updatedAt: currentIsoDate,
-    }));
-
-    await phacCollection.insertMany(phacItems);
-
-    const idMap = phacItems.reduce((a, v) => ({ ...a, [v.covid_id]: v.id }), {});
+    })));
+
+    const results = { successful: {}, duplicates: [], errors: [] };
+    try {
+      await phacCollection.insertMany(phacItems, { ordered: false });
+      results.successful = phacItems.reduce((a, v) => ({ ...a, [v.covid_id]: v.id }), {});
+    } catch (error) {
+      try {
+        const { writeErrors } = error.result.result;
+        results.duplicates = writeErrors
+          .filter((e) => e.err.code === 11000)
+          .map((e) => e.err.op.covid_id);
+        results.errors = writeErrors
+          .filter((e) => e.err.code !== 11000)
+          .map((e) => e.err.op.covid_id);
+        results.successful = phacItems.filter((i) => (
+          !results.duplicates.includes(i.covid_id) && !results.duplicates.includes(i.covid_id)
```

Comment:
Did you mean `!results.duplicates.includes(**i.id**) && !results.duplicates.includes(i.covid_id)` ?

Reply:
Whoops. Meant not in dupes and not in errors. Will fix.